tests/unit/models/employee-test.js

```diff
@@ -0,0 +1,105 @@
+import { mockUpdate, manualSetup, make, makeList } from 'ember-data-factory-guy';
+import { test, moduleForModel } from 'ember-qunit';
+import Ember from 'ember';
+const { run } = Ember;
+
+moduleForModel('employee', 'Unit | Model | employee', {
+  needs: [
+    'model:name',
+    'model:department-employment',
+    'model:department',
+    'model:nested-fragment/address',
+    'model:mailing-address',
+    'model:billing-address',
+    'transform:fragment',
+    'transform:fragment-array',
+    'transform:array'
+  ],
+
+  beforeEach: function() {
+    manualSetup(this.container);
+  }
+});
+
+// NAME FRAGMENT
+test('default employee', function() {
+  let employee = make('employee');
+  //Should I need a run loop?
+  run(() => {
+    ok(employee.get('name.firstName') === 'Tyrion');
+    ok(employee.get('name.lastName') === 'Lannister');
+  });
+});
+
+//MOCK UPDATE TEST
+test('update the employee', function() {
+  let employee = make('employee');
+  mockUpdate(employee);
+  run(() => {
+    ok(!employee.get('hasDirtyAttributes'));
+    employee.set('name.firstName', 'Jamie');
+    ok(employee.get('name.firstName') === 'Jamie');
+    ok(employee.get('name.lastName') === 'Lannister');
+    ok(employee.get('hasDirtyAttributes'));
+    employee.save();
+    ok(!employee.get('hasDirtyAttributes'));
```

Comment:
@danielspaniel I've just added this other failing unit test I've been getting. I'm not sure if it's related to `model-fragments` or not but if not I'll pull it out to a separate issue if you like.

Basically when doing a `mockUpdate` and then doing a `save()` the attributes are still dirty (is that expected?). Then test fails with the following error:

```
Promise rejected after update the employee: Assertion Failed: You can only unload a record which is not inFlight
```

which I'm presuming happens because the test is trying to clean up but the record is dirty.